ROM disassembler

danirod opened the pull request:
// TODO: Write

danirod committed:
Add initial prototype for ROM dumper

Will read the opcodes from a ROM and dump all the opcodes into
stdout. Still pending to process every opcode that is read.

danirod committed:
Ignore binary disassembler file

danirod committed:
Implement opcode decompiling

So WIP, missing a few things. Not well tested. Not tested at all.

danirod committed:
Fix Makefile.am

hugomartin89 commented on src/mdisasm.c:
```diff
@@ -1,6 +1,195 @@
+static void
+to_instruction_7(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "AND V[%x], %x", OPCODE_X(opcode), OPCODE_KK(opcode));
```
It should be ADD instead of AND

http://devernay.free.fr/hacks/chip8/C8TECH10.HTM#7xkk

hugomartin89 commented on src/mdisasm.c:
```diff
@@ -1,6 +1,195 @@
+to_instruction_D(unsigned short opcode, char* out)
+{
+    int x = OPCODE_X(opcode), y = OPCODE_Y(opcode), n = OPCODE_N(opcode);
+    snprintf(out, 20, "DRW %x, %x, %x", x, y, n);
```
registers instead constants (two first params)

http://devernay.free.fr/hacks/chip8/C8TECH10.HTM#Dxyn

hugomartin89 committed:
Minimal instruction fixes

Corrects 7xkk (ADD) and Dxyn (DRW).

danirod committed:
Merge pull request #30 from hugomartin89/fix/disasm

Fix some errors in mdisasm

danirod committed:
Use getopt for input parameters

Much much better

danirod committed:
Use different output modes

Using --output=full will render for each instruction, memory
address, compiled opcode and decompiled instruction. Using
--output=minimal will just render the decompiled instruction.
